R/kobo.R

```diff
@@ -63,21 +44,115 @@ Kobo <- R6::R6Class("Kobo",
         private$session_v2$get(path = paste0("api/v2/", path), query = query)
       } else if (version == "v1") {
         if (checkmate::test_null(private$session_v1)) {
-          usethis::ui_stop("Session for API v1 is not initalized. Please re-initalize the Kobo client with the base_url_v1 argument.")
+          usethis::ui_stop("Session for API v1 is not initalized.
+          Please re-initalize the Kobo client with the base_url_v1 argument.")
         }
         private$session_v1$get(path = paste0("api/v1/", path), query = query)
       } else {
         usethis::ui_stop(
           "Invalid version. Must be either v1 or v2.
-                    Come back in a couple of years."
+          Come back in a couple of years."
         )
       }
     },
+
+    #' @description
+    #' Wrapper for the POST method of internal session objects.
+    #' @param path character. Path component of the endpoint.
+    #' @param body R list or json-like string. A data payload to be sent
+    #' to the server.
+    #' @param version character. Indicates on which API version the request
+    #'  should be executed (available: `v1`, `v2`). Defaults to `v2`.
+    post = function(path, body, version = "v2") {
+      if (version == "v2") {
+        if (path != "imports/"){
+          private$session_v2$post(path = paste0("api/v2/", path), body = body)
+        } else {
+          private$session_v2$post(path = path, body = body)
+        }
+      } else if (version == "v1") {
+        if (checkmate::test_null(private$session_v1)) {
+          usethis::ui_stop("Session for API v1 is not initalized.
+          Please re-initalize the Kobo client with the base_url_v1 argument.")
+        }
+        private$session_v1$post(path = paste0("api/v1/", path), body = body)
+      } else {
+        usethis::ui_stop(
+          "Invalid version. Must be either v1 or v2.
+          Come back in a couple of years."
+        )
+      }
+    },
+
     #' @description
```

Comment:
An idea I had regarding the asset list / data:
Should we store them in a field of the `Kobo` object such that we can access it also from other methods? tbd

Reply:
I don't get. Could you please elaborate? :)

Reply:
I was just wondering whether we should store the asset list as a field of the Kobo object, that is, first declare a (private, or public) filed `asset_list`, and then inside this method include something like ` self$asset_list <- asset_list`  (or the same with private). 

Maybe this is something we could discuss? Just out of my gut, I thought this could be something useful. For example, linked to my comment on the clone asset method ("could we infer the asset type for the new asset from the type of the asset to be cloned?") could make use of such an internally stored asset list. 

But there are probably several other considerations, e.g. how often to update the internal asset list, etc. 

**TL;DR** I will try to elaborate my thoughts further until the next meeting, and we can ignore it for this PR.